docs/advanced_reference/developer_tooling/benchmarking.md

```diff
@@ -1,23 +1,34 @@
 
-1. Setup agent runtime environment in dev mode using `autonomy deploy build KEYS --dev`.
+1. A behaviour executes the `BenchmarkTool.save()` method.
+2. The agent service reaches the `reset_and_pause` round, and the Tendermint data is dumped.
+3. Stop the service execution after, at least, steps 1. and 2. have occurred in that sequence.
 
-2. Run agents for `1` period and wait until the round where the skill is configured to call the `BenchmarkTool.save` method. (You can also run agents for `N` periods if you want more data).
+The benchmarking data will be stored in the folder `<service_folder>/abci_build/persistent_data/benchmarks`.
 
-3. Point the aggregation script to the directory containing the benchmark data. This will generate a `benchmarks.html` file containing benchmark stats in your current directory.
+## Use the benchmarking tool
 
-## Example usage
+1. **Run the service.** Build and run the agent service. You can build and run the service either in [normal mode](../../guides/deploy_service.md#local-deployment) or in [dev mode](./dev_mode.md#build-and-run-an-agent-service-in-dev-mode).
 
-Run the `oracle/price_estimation` service as explained [here](../../demos/price_oracle_intro.md).
+2. **Wait for service execution.** Wait until the service has completed, at least, one period before cancelling the execution. That is, wait until the `reset_and_pause` round has occurred at least once. This is required because the Tendermint server will only dump Tendermint data when it reaches that state. Once you have a data dump, you can stop the local execution by pressing `Ctrl-C`.
 
-By default this will create a `4` agent runtime where you can wait until all `4` agents are at the end of the first period (you can wait for more periods if you want) and then you can stop the runtime. The data will be stored in the `abci_build/persistent_data/benchmarks` folder. You can use the following script to aggregate this data:
+3. **Aggregate the benchmarking data.** The benchmarking data will be stored in the folder `<service_folder>/abci_build/persistent_data/benchmarks`. Aggregate the data for all periods executing:
 
-```bash
-autonomy analyse benchmarks abci_build/persistent_data/benchmarks
-```
+    ```bash
+    autonomy analyse benchmarks abci_build/persistent_data/benchmarks
+    ```
+
+    This will generate a `benchmarks.html` file containing benchmark stats in your current directory.
+    By default the script will generate output for all periods but you can specify which period to generate output for. Similarly, block types aggregation is configurable as well. For example,
+
+    ```bash
+    autonomy analyse benchmarks abci_build/persistent_data/benchmarks --period 2 --block-type consensus
```

Comment:
provide information on the available block types and their use case prior to this.  It comes out of the blue and I have (supposedly) no idea yet what they are.